server/server.go

```diff
@@ -9,17 +9,67 @@ import (
 
 	"github.com/bombsimon/amqp-rpc/connection"
 	"github.com/bombsimon/amqp-rpc/logger"
-	"github.com/bombsimon/amqp-rpc/middleware"
+)
+
+type ctxKey string
```

Comment:
Is this necessary? I saw the only reference to it's value required a type casting to a string.

Reply:
Yep, because golint complains if the context keys are just strings, this was added a while ago:

https://github.com/golang/go/issues/17293
https://github.com/golang/lint/commit/c6242afa6ced3be489e1184eb80bc2d85f1f5e7b#diff-f90a83eaf70c284b08d2dc4bcc941d13R13